monkey/infection_monkey/network/scan_target_generator.py

```diff
@@ -0,0 +1,131 @@
+            inaccessible_subnets, local_network_interfaces
+        )
+        scan_targets.update(inaccessible_subnets)
+
+    _remove_interface_ips(scan_targets, local_network_interfaces)
+    _remove_blocklisted_ips(scan_targets, blocklisted_ips)
+
+    scan_target_list = list(scan_targets)
+    scan_target_list.sort()
+
+    return scan_target_list
+
+
+def _get_ips_from_ranges_to_scan(ranges_to_scan: List[str]) -> Set[str]:
+    scan_targets = set()
+
+    ranges_to_scan = _filter_invalid_ranges(
+        ranges_to_scan, "Bad network range input for targets to scan:"
+    )
+
+    network_ranges = [NetworkRange.get_range_obj(_range) for _range in ranges_to_scan]
+    for _range in network_ranges:
+        scan_targets.update(set(_range))
+    return scan_targets
+
+
+def _get_ips_to_scan_from_local_interface(interfaces: List[NetworkInterface]) -> Set[str]:
+    ranges = [f"{interface.address}{interface.netmask}" for interface in interfaces]
+
+    ranges = _filter_invalid_ranges(ranges, "Local network interface returns an invalid IP:")
+    return _get_ips_from_ranges_to_scan(ranges)
+
+
+def _remove_interface_ips(scan_targets: Set[str], interfaces: List[NetworkInterface]):
+    interface_ips = [interface.address for interface in interfaces]
+    _remove_ips_from_scan_targets(scan_targets, interface_ips)
+
+
+def _remove_blocklisted_ips(scan_targets: Set[str], blocked_ips: List[str]):
+    filtered_blocked_ips = _filter_invalid_ranges(blocked_ips, "Invalid blocked IP provided:")
+    if not len(filtered_blocked_ips) == len(blocked_ips):
+        raise InvalidNetworkRangeError("Received an invalid blocked IP. Aborting just in case.")
+    _remove_ips_from_scan_targets(scan_targets, blocked_ips)
+
+
+def _remove_ips_from_scan_targets(scan_targets: Set[str], ips_to_remove: List[str]):
+    for ip in ips_to_remove:
+        try:
+            scan_targets.remove(ip)
+        except KeyError:
+            # We don't need to remove the ip if it's already missing from the scan_targets
+            pass
+
+
+def _get_segmentation_check_targets(
+    inaccessible_subnets: List[str], local_interfaces: List[NetworkInterface]
+):
+    subnets_to_scan = set()
+    local_ips = [interface.address for interface in local_interfaces]
+
+    local_ips = _filter_invalid_ranges(local_ips, "Invalid local IP found: ")
+    inaccessible_subnets = _filter_invalid_ranges(
+        inaccessible_subnets, "Invalid segmentation scan target: "
+    )
+
+    inaccessible_subnets = _convert_to_range_object(inaccessible_subnets)
+    subnet_pairs = itertools.product(inaccessible_subnets, inaccessible_subnets)
+
+    for (subnet1, subnet2) in subnet_pairs:
+        if _is_segmentation_check_required(local_ips, subnet1, subnet2):
+            ips = _get_ips_from_ranges_to_scan(subnet2)
+            subnets_to_scan.update(ips)
+
+    return subnets_to_scan
+
+
+def _filter_invalid_ranges(ranges: List[str], error_msg: str) -> List[str]:
+    filtered = []
```

Comment:
`valid_ranges` is a noun. A method should be a verb. `filter` expresses that we get a number of items and remove some of them, while keeping others: "to remove by means of a filter". Maybe `_remove_invalid_ranges`, but I think `filter` is better than `remove`, because it returns the filtered out list

Reply:
I meant the local variable defined on line 106. "filtered" is an ambiguous name for it. "valid_ranges" is clearer.